src/test/java/converterTest/ConverterTest.java

```diff
@@ -3,121 +3,112 @@
-						scale ("25.75", SCALE), scale ("-35", SCALE))),
+				() -> assertThrows (InvalidAmountOrExchangeRateException.class, () -> converter.conversionUSDtoEUR (
+						new BigDecimal("25.75"), new BigDecimal("-35"))),
 
-				() -> assertThrows (AmountOrExchangeException.class, () -> converter.conversionUSDtoEUR (
-						scale ("45.73", SCALE), scale ("100", SCALE))),
+				() -> assertThrows (InvalidAmountOrExchangeRateException.class, () -> converter.conversionUSDtoEUR (
+						new BigDecimal("45.73"), new BigDecimal("100"))),
 
-				() -> assertThrows (AmountOrExchangeException.class, () -> converter.conversionUSDtoEUR (
-						scale ("55.73", SCALE), scale ("100.01", SCALE))),
+				() -> assertThrows (InvalidAmountOrExchangeRateException.class, () -> converter.conversionUSDtoEUR (
+						new BigDecimal("55.73"), new BigDecimal("100.01"))),
 
-				() -> assertThrows (AmountOrExchangeException.class, () -> converter.conversionUSDtoEUR (
-						null, scale ("11", SCALE))),
+				() -> assertThrows (InvalidAmountOrExchangeRateException.class, () -> converter.conversionUSDtoEUR (
+						null, new BigDecimal("11"))),
 
-				() -> assertThrows (AmountOrExchangeException.class, () -> converter.conversionUSDtoEUR (
-						scale ("12.73", SCALE), null))
+				() -> assertThrows (InvalidAmountOrExchangeRateException.class, () -> converter.conversionUSDtoEUR (
+						new BigDecimal("12.73"), null))
 		);
 	}
 
 	@Test
 	@DisplayName("conversionEURtoUSD() normal conversion should work")
 	void conversionEURtoUSD_Test() {
 		assertAll ("conversionEURtoUSD() normal conversion",
-				() -> assertEquals(scale ("13.70", SCALE),
-						converter.conversionEURtoUSD (scale ("15.11", SCALE), scale ("10.23", SCALE))),
+				() -> assertEquals(new BigDecimal("13.70"),
+						converter.conversionEURtoUSD (new BigDecimal("15.11"), new BigDecimal("10.23"))),
 
-				() -> assertEquals(scale ("7.00", SCALE),
-						converter.conversionEURtoUSD (scale ("7.15", SCALE), scale ("3", SCALE))),
+				() -> assertEquals(new BigDecimal("7.00"),
+						converter.conversionEURtoUSD (new BigDecimal("7.15"), new BigDecimal("3"))),
 
-				() -> assertEquals(scale ("4.83", SCALE),
-						converter.conversionEURtoUSD (scale ("5", SCALE), scale ("4.32", SCALE))),
+				() -> assertEquals(new BigDecimal("4.83"),
+						converter.conversionEURtoUSD (new BigDecimal("5"), new BigDecimal("4.32"))),
 
-				() -> assertEquals(scale ("8.59", SCALE),
-						converter.conversionEURtoUSD (scale ("10", SCALE), scale ("15", SCALE)))
+				() -> assertEquals(new BigDecimal("8.59"),
+						converter.conversionEURtoUSD (new BigDecimal("10"), new BigDecimal("15")))
 		);
 	}
 
 	@Test
 	@DisplayName("conversionEURtoUSD() Exception should work")
 	void conversionEURtoUSD_ExceptionTest() {
 		assertAll ("conversionEURtoUSD() Exceptions",
-				() -> assertThrows (AmountOrExchangeException.class, () -> converter.conversionEURtoUSD (
```

Comment:
Now some tests are failing. But this is actually a good think because they should check the code behaviour. So this error
```
org.opentest4j.AssertionFailedError: 
Expected :7.7248
Actual   :7.72
```
actually means you have to fix the code, probably just set the right scale for method output